### ISSUE TITLE
Q-Chem: `StopIteration` is raised for incomplete SCF

### ISSUE THREAD

**Issue by berquist** [cclib/cclib#206]
Any time an SCF is incomplete (*not* unconverged), the parser doesn't handle it gracefully:

```
~/calc.sgr/krista/batch2/jobs $ l *out
/home/dlambrecht/erb74/calc.sgr/krista/batch2/jobs
-rw-r----- 1 erb74 dlambrecht 825K Aug 10 19:02 CO2_BMIM_PF6.b3lyp.vqz.opt.out
-rw-r----- 1 erb74 dlambrecht 1.5M Aug 10 19:07 CO2_BMIM_TFA.blyp.avtz.opt.out
~/calc.sgr/krista/batch2/jobs $ tail *out
==> CO2_BMIM_PF6.b3lyp.vqz.opt.out <==
 -------------------------------------------------------
 OpenMP BLAS3 based DFT computing Module                
 Release: version 1.0, May 2013, Q-Chem Inc. Pittsburgh 
 -------------------------------------------------------
 ---------------------------------------
  Cycle       Energy         DIIS Error
 ---------------------------------------
    1   -1553.2043340273      1.36E-05
    2   -1553.2044609529      7.13E-07
    3   -1553.2044610939      5.17E-07

==> CO2_BMIM_TFA.blyp.avtz.opt.out <==
Setting SaveMinima to 0
 using 8 threads for integral computing 
 -------------------------------------------------------
 OpenMP BLAS3 based DFT computing Module                
 Release: version 1.0, May 2013, Q-Chem Inc. Pittsburgh 
 -------------------------------------------------------
 ---------------------------------------
  Cycle       Energy         DIIS Error
 ---------------------------------------
    1   -1138.3465451227      8.10E-05
~/calc.sgr/krista/batch2/jobs $ cclib_plot_geometry_convergence.py *.out
Traceback (most recent call last):
  File "/home/dlambrecht/erb74/scripts/cclib_plot_geometry_convergence.py", line 76, in <module>
    main()
  File "/home/dlambrecht/erb74/scripts/cclib_plot_geometry_convergence.py", line 38, in main
    data = job.parse()
  File "/home/dlambrecht/software/python/anaconda3/lib/python3.4/site-packages/cclib/parser/logfileparser.py", line 247, in parse
    self.extract(inputfile, line)
  File "/home/dlambrecht/software/python/anaconda3/lib/python3.4/site-packages/cclib/parser/qchemparser.py", line 347, in extract
    line = next(inputfile)
  File "/home/dlambrecht/software/python/anaconda3/lib/python3.4/site-packages/cclib/parser/logfileparser.py", line 66, in __next__
    return self.next()
  File "/home/dlambrecht/software/python/anaconda3/lib/python3.4/site-packages/cclib/parser/logfileparser.py", line 61, in next
    line = next(self.file)
StopIteration
```

**Comment by langner** [comment]
Do you have a fix for this?

**Comment by berquist** [comment]
Yes, but I'm waiting on my other PR to go through, for two reasons:

1. That one changes the indentation of almost the entire `extract()` method and I don't know how git will handle the merge.
2. I've noticed that at least one other parser (NWChem) doesn't handle this properly either, so I want to check all the parsers and combine any changes to them into one PR.

**Comment by langner** [comment]
> Yes, but I'm waiting on my other PR to go through

Which PR is that?

**Comment by berquist** [comment]
> Which PR is that?

It was https://github.com/cclib/cclib/pull/236 and https://github.com/cclib/cclib-data/pull/20.

Here is the PR that fixes this issue: https://github.com/cclib/cclib/pull/250

**Comment by langner** [comment]
Fixed by #250
